### PR TITLE
scx_layered: Fix verifier issues on older kernels

### PR DESCRIPTION
**Pull request by hodgesds**
On 6.9 kernels the verifier is not able to track `struct bpf_cpumasks` properly on nested structs. Move the cpumasks from the `cached_cpus` struct back to the `task_ctx` struct so older versions of the verifier can pass.

### CODE REVIEW

File: scheds/rust/scx_layered/src/bpf/main.bpf.c [hunk 1]
```diff
@@ -1564,7 +1578,8 @@ static __noinline int consume_non_open(struct cost *costc, s32 cpu, u32 my_llc_i
 					return -EINVAL;
 				}
 				layer = MEMBER_VPTR(layers, [layer_idx]);
-				if (has_budget(costc, layer) == 0)
+				if (layer->kind == LAYER_KIND_OPEN ||
```
Comment (etsal) [inline scheds/rust/scx_layered/src/bpf/main.bpf.c:1581]: Is this related to the main change of the diff?

**Review by etsal — APPROVED** [review]

File: scheds/rust/scx_layered/src/bpf/main.bpf.c (hunk 1 above)
Comment (hodgesds) [reply scheds/rust/scx_layered/src/bpf/main.bpf.c:1581]: I can pull this into a separate diff, good call.

**Review by htejun — APPROVED** [review]
